packages/expect/src/utils.ts

```diff
@@ -16,3 +22,26 @@ export function recordAsyncExpect(test: any, promise: Promise<any> | PromiseLike
 
   return promise
 }
+
+export function wrapSoft(utils: Chai.ChaiUtils, fn: (this: Chai.AssertionStatic & Assertion, ...args: any[]) => void) {
+  return function (this: Chai.AssertionStatic & Assertion, ...args: any[]) {
+    const test: Test = utils.flag(this, 'vitest-test')
+
+    // @ts-expect-error local is untyped
+    const state: MatcherState = test?.context._local
+      ? test.context.expect.getState()
+      : getState((globalThis as any)[GLOBAL_EXPECT])
+
+    if (!test || !state.soft)
+      return fn.apply(this, args)
+
+    try {
+      return fn.apply(this, args)
+    }
+    catch (err) {
+      test.result!.state = 'fail'
```

Comment:
We should probably add a guard here `test.result ??= {}` instead of suppressing errors

Reply:
I think it's nice to have for custom implementations